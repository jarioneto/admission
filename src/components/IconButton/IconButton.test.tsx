import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { IconButton, IconButtonProps } from './index';
import { HiUserAdd } from 'react-icons/hi';

const renderComponent = (props: IconButtonProps = {}) =>
  render(<IconButton {...props}><HiUserAdd /></IconButton>);

describe('IconButton component', () => {
  it('should render the component', () => {
    renderComponent();

    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('should trigger on click when pressed', async () => {
    const props = {
      onClick: jest.fn(),
    };

    renderComponent(props);

    const button = screen.getByRole('button');
    await userEvent.click(button);

    expect(props.onClick).toHaveBeenCalled();
  });

  it('should do not trigger on click if disabled when pressed', async () => {
    const props = {
      disabled: true,
      onClick: jest.fn(),
    };

    renderComponent(props);

    const button = screen.getByRole('button');
    await userEvent.click(button);

    expect(props.onClick).not.toHaveBeenCalled();
  });

  describe.each`
    variant
    ${'solid'}
    ${'outlined'}
  `('$variant', ({ variant }) => {
    it(`should render ${variant} button`, () => {
      const { container } = renderComponent({ variant });
      expect(container).toMatchSnapshot();
    });
  });

  describe.each`
    color
    ${'primary'}
    ${'secondary'}
    ${'tertiary'}
  `('$color', ({ color }) => {
    it(`should render ${color} button`, () => {
      const { container } = renderComponent({ color });
      expect(container).toMatchSnapshot();
    });
  });

  describe.each`
    shape
    ${'rectangle'}
    ${'circle'}
    ${'pill'}
  `('$shape', ({ shape }) => {
    it(`should render ${shape} button`, () => {
      const { container } = renderComponent({ shape });
      expect(container).toMatchSnapshot();
    });
  });
});
