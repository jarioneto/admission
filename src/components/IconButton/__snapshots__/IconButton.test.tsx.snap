// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component circle should render circle button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB YWqnm sc-gEvDqW hJYzlF"
    color="primary"
    shape="circle"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component outlined should render outlined button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB YWqnm sc-gEvDqW hJYzlF"
    color="primary"
    shape="circle"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component pill should render pill button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB jFtxKj sc-gEvDqW drzIsX"
    color="primary"
    shape="pill"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component primary should render primary button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB YWqnm sc-gEvDqW hJYzlF"
    color="primary"
    shape="circle"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component rectangle should render rectangle button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB juLFoy sc-gEvDqW drzIsX"
    color="primary"
    shape="rectangle"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component secondary should render secondary button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB hcWssO sc-gEvDqW hJYzlF"
    color="secondary"
    shape="circle"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component solid should render solid button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB jZJIRv sc-gEvDqW hJYzlF"
    color="primary"
    shape="circle"
    variant="solid"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton component tertiary should render tertiary button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB bznnSW sc-gEvDqW hJYzlF"
    color="tertiary"
    shape="circle"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 9a3 3 0 100-6 3 3 0 000 6zM8 11a6 6 0 016 6H2a6 6 0 016-6zM16 7a1 1 0 10-2 0v1h-1a1 1 0 100 2h1v1a1 1 0 102 0v-1h1a1 1 0 100-2h-1V7z"
      />
    </svg>
  </button>
</div>
`;
